src/main/java/com/github/lookout/metrics/agent/generators/YammerMetricsGenerator.java

```diff
@@ -0,0 +1,87 @@
+package com.github.lookout.metrics.agent.generators;
+
+import com.timgroup.statsd.StatsDClient;
+import com.yammer.metrics.Metrics;
+import com.yammer.metrics.core.*;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Map;
+import java.util.Set;
+import java.util.SortedMap;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Created by rkuris on 5/28/15.
+ */
+public class YammerMetricsGenerator implements MetricProcessor<Long>, MetricGenerator {
+
+    private static final Logger LOG = LoggerFactory.getLogger(YammerMetricsGenerator.class);
+    private MetricsRegistry registry = Metrics.defaultRegistry();
+    private StatsDClient client;
+    private static final MetricPredicate predicate = MetricPredicate.ALL;
+    protected final Clock clock = Clock.defaultClock();
+
+    @Override
+    public void generate(StatsDClient client) {
+        // get all the registered metrics via dropwizard
+        this.client = client;
+        final long epoch = TimeUnit.MILLISECONDS.toSeconds(clock.time());
+        final Set<Map.Entry<String, SortedMap<MetricName, Metric>>> entries = registry.groupedMetrics(predicate).entrySet();
+
+        for (final Map.Entry<String, SortedMap<MetricName, Metric>> entry : entries) {
+            for (final Map.Entry<MetricName, Metric> subEntry : entry.getValue().entrySet()) {
+
+                final Metric metric = subEntry.getValue();
+
+                if (metric != null) {
+                    try {
+                        metric.processWith(this, subEntry.getKey(), epoch);
+                    } catch (final Exception exception) {
+                        LOG.error("Error processing key {}", subEntry.getKey(), exception);
+                    }
+                }
+            }
+        }
+
+    }
+    @Override
+    public void processMeter(MetricName name, Metered meter, Long epoch) throws Exception {
+        LOG.debug("Meter {} {} skipped", name.getName(), meter.count());
+    }
+
+    @Override
+    public void processCounter(MetricName name, Counter counter, Long epoch) throws Exception {
+        client.gauge(name.getName(), counter.count());
```

Comment:
There are a lot of good questions here, which I don't have good answers to. One is, do we care which mbean produced the statistic from DataDog? My initial take on that is 'no' because that's an implementation detail as to which module produced the stat.

Another good question is: is there any chance of duplicated values? I think the answer is yes, but I think that would be really confusing for the consumer of JMX to understand the differences, unless they are qualified by something (like per-table stats).

And then there's the dashboard, and how tricky it might be to type in or find the appropriate variable. It actually isn't that bad, except that the default graph name with a long attribute doesn't look as nice as a short one.